src/installer/image/unpack.go

```diff
@@ -0,0 +1,135 @@
+package image
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/Dynatrace/dynatrace-operator/src/installer/zip"
+	"github.com/containers/image/v5/copy"
+	"github.com/containers/image/v5/manifest"
+	"github.com/containers/image/v5/signature"
+	"github.com/containers/image/v5/types"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/afero"
+)
+
+type imagePullInfo struct {
+	imageCacheDir  string
+	targetDir      string
+	sourceCtx      *types.SystemContext
+	destinationCtx *types.SystemContext
+	sourceRef      *types.ImageReference
+	destinationRef *types.ImageReference
+}
+
+func (installer *imageInstaller) getAgentFromImage(pullInfo imagePullInfo) error {
+	manifestBlob, err := copyImageToCache(pullInfo)
+	if err != nil {
+		log.Info("failed to get manifests blob",
+			"image", installer.props.Image,
+		)
+		return err
+	}
+
+	manifests, err := installer.unmarshalManifestBlob(manifestBlob, pullInfo.imageCacheDir)
+	if err != nil {
+		log.Info("failed to unmarshal manifests",
+			"image", installer.props.Image,
+			"manifestBlob", manifestBlob,
+			"imageCacheDir", pullInfo.imageCacheDir,
+		)
+		return err
+	}
+	return installer.unpackOciImage(manifests, pullInfo.imageCacheDir, pullInfo.targetDir)
+
+}
+
+func (installer *imageInstaller) unmarshalManifestBlob(manifestBlob []byte, imageCacheDir string) ([]*manifest.OCI1, error) {
+	var manifests []*manifest.OCI1
+
+	mimeType := manifest.GuessMIMEType(manifestBlob)
+
+	if mimeType == ocispec.MediaTypeImageManifest {
+		ociManifest, err := manifest.OCI1FromManifest(manifestBlob)
+		if err != nil {
+			return manifests, err
+		}
+		manifests = append(manifests, ociManifest)
+
+	} else if mimeType == ocispec.MediaTypeImageIndex {
+		ociManifests, err := unmarshallImageIndex(installer.fs, imageCacheDir, manifestBlob)
+		if err != nil {
+			return nil, err
+		}
+		manifests = append(manifests, ociManifests...)
+	}
```

Comment:
An implementation that does not need to be changed that much when a new mime type is added would be preferred.

Maybe a map of functions to be called on specific types?